### DISCUSSION TITLE
Challenges of Negative Sampling in Bipartite Graphs

### DISCUSSION THREAD

**Discussion by nemat-gholinejad** [pyg-team/pytorch_geometric#7798] [General]
Hello,
When working with the negative sampling part and the necessity to use the function "structured_negative_sampling" there might be inaccuracies when dealing with a bipartite graph. This issue becomes more evident if the number of nodes on the right side of the bipartite graph exceeds the number of nodes on the left side. For instance, consider the movie dataset, which consists of 6000 users and 4000 items, often employed in link prediction and recommender systems.
I'm uncertain whether adding this feature falls within the scope of this library. However, I bring it up because, in the contribution section, you mentioned that we should address any uncertainties with you.

**Comment by akihironitta** [comment]
#7799 for reference
